sys/fido2/doc.txt

```diff
@@ -0,0 +1,209 @@
+ *
+ * ### General
+ *
+ * Following is an overview of the entities of this implementation and their relationships:
+ *
+ *
+ *                     +-----------+  +-----------+   +-----------+   +-----------+
+ *                     |           |  |           |   |           |   |           |
+ *                     | ctap_cbor |  |ctap_crypto|   | ctap_mem  |   |ctap_utils |
+ *                     |           |  |           |   |           |   |           |
+ *                     +-----------+  +-----------+   +-----------+   +-----|-----+
+ *                           |             |               |                |
+ *                           |             |               |                |
+ *                           |         +---------------------------+        |
+ *                           |         |                           |        |
+ *                           |---------|           ctap            |--------|
+ *                                     |                           |
+ *                                     +---------------------------+
+ *                                                   |
+ *                                     +-------------|-------------+
+ *                                     |                           |
+ *                                     |       ctap_transport      |
+ *                                     |                           |
+ *                                     +---------------------------+
+ *                                           |
+ *                                           |
+ *                                     +-----------+
+ *                                     |           |
+ *                                     | ctap_hid  |
+ *                                     |           |
+ *                                     +-----------+
+ *
+ *
+ *
+ * **ctap_hid**
+ *
+ * USB Human Interface Device (USB HID) transport binding for CTAP (CTAPHID).
+ *
+ * Initializes the USBUS HID interface.
+ *
+ * Communicates with the USBUS USB HID interface through the USBUS HID IO interface.
+ *
+ * **ctap_transport**
+ *
+ * Initializes CTAP layer.
+ *
+ * Initializes CTAP event queue.
+ *
+ * Manages CTAP transport bindings (currently only CTAPHID).
+ *
+ * **ctap**
+ *
+ * Contains the main CTAP logic.
+ *
+ * Makes use of helpers for flash access, cryptographic operations and CBOR operations.
+ *
+ * **ctap_cbor**
+ *
+ * Helper containing functionality to parse and encode CBOR messages. Uses the [tinyCBOR](https://doc.riot-os.org/group__pkg__tinycbor.html) pkg.
+ *
+ * **ctap_crypto**
+ *
+ * Helper containing functionality for cryptographic operations.
+ *
+ * Abstraction for native cryptographic operations (SHA256, HMAC-SHA-256, AES CCM).
+ *
+ * @warning This abstraction ignores the pointer to a static buffer returned by some functions (e.g. @ref sha256). Therefore a valid pointer to a buffer for the digest always needs to be provided when necessary.
```

Comment:
I don't quite understand what you are trying to say here. The key point is: The abstraction exposes error return values which are currently not used in all cases. This should be noted